sdk/core/azure-core/azure/core/polling/_poller.py

```diff
@@ -203,7 +206,7 @@ def wait(self, timeout=None):
         self._thread.join(timeout=timeout)
         try:
             # Let's handle possible None in forgiveness here
-            raise self._exception  # type: ignore
+            raise cast(Exception, self._exception)
```

Comment:
Cast is used to clarify the type when Mypy is lost, here it's not lost, he correctly detects that it can be `None`, and that we can't raise `None`. We actually plan that it could be `None`, and it's why we catch a `TypeError` just below. Mypy is not smart enough to understand that, see issue https://github.com/python/mypy/issues/8165

I'm fine with a type ignore and a ref to that bug, but a cast is a mis-use here